driver-async/src/main/com/mongodb/async/client/AbstractSubscription.java

```diff
@@ -120,20 +123,23 @@ void addToQueue(@Nullable final List<TResult> results) {
     void onError(final Throwable t) {
         if (terminalAction()) {
             postTerminate();
-            observer.onError(t);
+            try {
+                observer.onError(t);
+            } catch (Throwable t1) {
+                LOGGER.error("Calling onError threw an exception", t1);
+                throw MongoException.fromThrowableNonNull(t1);
+            }
+        } else {
+            throw MongoException.fromThrowableNonNull(t);
```

Comment:
If the Observer has terminated already, still throw the exception.   This is done to ensure errors surfacing in `onNext` and `onComplete` are thrown.